### PR TITLE
Bind crossplane rbac-manager edit role to customer's cluster admin

### PR DESCRIPTION
**Pull request by piontec**
This PR adds a new crossplane specific controller, `crossplaneauth`. It detects when a ClusterRole `crossplane-edit` is created/deleted (which means crossplane is [un]installed with its `rbac-manager`) and binds `default:automation` SA and customer's admin user group to this ClusterRole.

**DO NOT RELEASE** before [new crossplane](https://github.com/giantswarm/crossplane/pull/19) version is released.

## Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] If the change introduces new roles, please consider adding user-friendly descriptions to the roles with the annotation [giantswarm.io/notes](https://github.com/giantswarm/k8smetadata/blob/dff3b2358977e13c90479c66e21c728a96ddfe6d/pkg/annotation/general.go#L12) to help explain their purpose and usage.

### CODE REVIEW

File: service/controller/crossplane/key/key.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	if !ok {
+		return rbacv1.ClusterRole{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", p, v)
+	}
+
+	c := p.DeepCopy()
+
+	return *c, nil
+}
+
+func GetClusterRoleBindingName() string {
```
Comment (anvddriesch) [inline service/controller/crossplane/key/key.go:23]: Any reason not to make these two functions just string constants?

File: service/controller/crossplane/resource/crossplaneauth/create.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			Kind:       "ClusterRoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: key.GetClusterRoleBindingName(),
+			Labels: map[string]string{
+				label.ManagedBy: project.Name(),
+			},
+			Annotations: map[string]string{
+				annotation.Notes: "Grants customer's cluster-admin permissions to use crossplane rbac-manager manged crossplane:edit ClusterRole",
```
Comment (anvddriesch) [inline service/controller/crossplane/resource/crossplaneauth/create.go:56]:
```suggestion
				annotation.Notes: "Grants customer's cluster-admin permissions to use crossplane rbac-manager managed crossplane:edit ClusterRole",
```

**Review by anvddriesch — APPROVED** [review]
LGTM, just small nitpicks :)

File: service/controller/crossplane/key/key.go (hunk 1 above)
Comment (piontec) [reply service/controller/crossplane/key/key.go:23]: Yeah, I'm working on making them a config option. Will be improved in the next PR.

File: service/controller/crossplane/crossplane.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var clusterRoleAuthController *controller.Controller
+	{
+		c := controller.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+			NewRuntimeObjectFunc: func() client.Object {
```
Comment (ljakimczuk) [inline service/controller/crossplane/crossplane.go:45]: Does Crossplane's RBAC Manager use any label which could be used here for selecting resources, to narrow down the scope?

Comment (piontec) [reply service/controller/crossplane/crossplane.go:46]: They do, I was just afraid that if they change the labels at any point, and we won't notice, the binding will stop working. The same applies to ClusterRole name, but I expect this to not happen that easily. Do you think we have enough ClusterRoles for that to become a problem?

We can use the following labels:

```
  labels:
    app: crossplane
    app.kubernetes.io/component: cloud-infrastructure-controller
    app.kubernetes.io/instance: crossplane-crossplane
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: crossplane
    app.kubernetes.io/part-of: crossplane
```

**Review by kubasobon — APPROVED** [review]

**Pull request merged**